test/Aqua.jl

```diff
@@ -0,0 +1,10 @@
+module AquaTests
+
+using Aqua: Aqua
+using Turing
+
+# TODO(mhauru) We skip testing for method ambiguities because it catches a lot of problems
+# in dependencies. Would like to check it for just Turing.jl itself though.
+Aqua.test_all(Turing; ambiguities=false)
```

Comment:
If you want to test ambiguities, I suggest
# Test ambiguities separately without Base and Core
```suggestion
@testset "Aqua" begin
    # Test ambiguities separately without Base and Core
    # Ref: https://github.com/JuliaTesting/Aqua.jl/issues/77
    Aqua.test_all(Turing; ambiguities = false)
    Aqua.test_ambiguities(Turing)
end
```

Reply:
I won't get to this today, and next week I'm off this project, so I'll merge what's here now and open an issue to come back to method ambiguities.